src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  // Tablica, która będzie przechowywać historię stanów
+  const stateHistory = [];
+
+  // Zaczynamy od klonowania początkowego stanu
+  let currentState = { ...state };
```

Comment:
[NAMING] - The variable name for the object copy should be more descriptive to reflect its purpose. Instead of `currentState`, a name like `stateCopy` would be more appropriate and in line with good naming conventions.